karapace/kafka_rest_apis/consumer_manager.py

```diff
@@ -34,7 +34,16 @@ def new_name() -> str:
 class ConsumerManager:
     def __init__(self, config: dict) -> None:
         self.config = config
-        self.hostname = f"http://{self.config['advertised_hostname']}:{self.config['port']}"
+        if self.config["advertised_port"] is None:
+            self.hostname = (
+                f"{self.config['advertised_protocol']}://"
+                f"{self.config['advertised_hostname']}:{self.config['port']}"
+            )
```

Comment:
To fix the linter complain:
```suggestion
            self.hostname = (
                f"{self.config['advertised_protocol']}://{self.config['advertised_hostname']}:{self.config['port']}"
            )
```

---

As a note, I recommend getting `pre-commit` installed on your git repo:

https://pre-commit.com/#usage

That way the tool will run locally and you will get faster feedback cycles

Reply:
Thanks a lot! I have installed and tried the pre-commit, it gave me exactly the same suggestion you sent. After changing it, it passed on my local. Appreciate that, will commit the change very soon.

Reply:
Thank you for the contribution :)